24.md

```diff
@@ -0,0 +1,57 @@
+NIP-24
+======
+
+Protect the Unknown
+-------------------
+
+`draft` `optional` `client` `author:giszmo`
+
+## Replaceable Events
+
+A *replaceable event* like `kind: 0`, `kind: 3` and as defined in
+[NIP-16](16.md).
+
+## Client Behavior
+
+Clients MUST NOT change or delete Data they do not understand without user
+confirmation.
+
+Rationale
+---------
+
+### Different extensions to a feature
+
+Clients differ in their feature sets and as nips expand kinds, clients that do
+not consider unsupported parameters remove those when "replacing" events,
+breaking the clients' compatibility.
+
+Alice uses Branle at home and Damus on mobile. She configured `nip05` on Branle
+but when storing her `screen_name` on Damus using the `kind: 0` metadata, her
+`nip05` configuration disappeard. When configuring this on Damus again, her
+`screen_name` disappeared from her metadata.
+
+Bob's client invited him to define [NIP-02](02.md) petnames. And so he did. For
+all 200 follows. On his different mobile client he added his follow 201 and all
+petnames were gone.
+
+### Different features using the same `kind`
+
+There will be collisions between kinds where data stored in a replaceable event
+by one client means something completely different than what another client
+would want to store there.
+
+Carol found this novel dating website and configured her (nostr `kind: 10101`)
+profile. Some time later she installed that game using her common identity
+thanks to [NIP-07](07.md) and all her dating app configuration was gone.
+
+Implementation
+--------------
+
+Nostr does not provide a means by which a client may know it has seen the event
+it is replacing, resulting in a theoretical impossibility of guaranteeing this
+NIP's promise but by advertising the client to support this NIP, it promises to
+at least attempt to avoid interference with other apps using the same `kind`s.
+
+At the very minimum, a client should not drop data that itself does not use, so
```

Comment:
could we add subscribing to existing events to the minimum requirements? like:

At the very minimum, a client should:
- subscribe to existing events of the relevant kind
- not drop data that itself does not use, so
instead of only serializing its own client data, it should update the generic
JSON object found in `content` or Array<Array<String>> in `tags`